### PR TITLE
Preserve the compression metadata in copyObject with REPLACE

### LINKED ISSUE #6951: Compression: CopyObject strips Compression related headers, if metadata is passed
Copyobject seems to strip compression related headers if metadata is passed

## Expected Behavior
Compression related headers should not be removed.

## Current Behavior
Compression related headers are removed.

## Steps to Reproduce (for bugs)
1. Start minio for FS or XL with compression enabled
2. Create a bucket called `testbucket`
3. Create an object called `object_copy.txt`
4. Then execute the following code
```
from minio import Minio
from minio.error import ResponseError

client = Minio('localhost:9000',
               access_key='minio',
               secret_key='minio123',
	       secure=False)

# Make a new bucket
try:
    client.copy_object('testbucket', 'object_copy.txt','/testbucket/object_source.txt',metadata={"testing-string": "string", "testing-int": 1})
except ResponseError as err:
    print(err)
```
5. Then do mc stat on both the source and the destination files, you can see that the size is the compressed object size. 

6. If you look in the backend, you can see that the compression related headers have been removed in `fs.json` for the target file.

## Context
Saw failures in minio-py test in mint.

## Regression
Don't know

## Your Environment

* Version used (`minio version`):
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

### PR DESCRIPTION
**Pull request by Praveenrajmani**
Compression related metadata was overridden by the source's metadata with the REPLACE option

## Description
Preserve the metadata and store it after

## Motivation and Context
Fixes #6951

## Regression
No

## How Has This Been Tested?
```
from minio import Minio
from minio.error import ResponseError

client = Minio('localhost:9000',
               access_key='minio',
               secret_key='minio123',
	       secure=False)

# Make a new bucket
try:
    client.copy_object('testbucket', 'object_copy.txt','/testbucket/object_source.txt',metadata={"testing-string": "string", "testing-int": 1})
except ResponseError as err:
    print(err)

```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

File: cmd/object-handlers.go [hunk 1]
```diff
@@ -788,13 +788,14 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		length = actualSize
 	}
 
+	var compressMetadata = make(map[string]string)
```
Comment (harshavardhana) [inline cmd/object-handlers.go]: you only need to allocate this if `isCompressed` is true so start with 

```
var compressMetadata map[string]string

if isCompressed {
    compressMetadata = make(map[string]string)
}

for k, v := range compressMetadata {
}
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/object-handlers.go (hunk 1 above)
Comment (Praveenrajmani) [reply cmd/object-handlers.go]: Done :+1:

File: cmd/object-handlers.go [hunk 2]
```diff
@@ -788,13 +788,15 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
...
 
+	var compressMetadata map[string]string
 	// No need to compress for remote etcd calls
 	// Pass the decompressed stream to such calls.
 	isCompressed := objectAPI.IsCompressionSupported() && isCompressible(r.Header, srcObject) && !isRemoteCallRequired(ctx, srcBucket, dstBucket, objectAPI)
 	if isCompressed {
-		// Storing the compression metadata.
-		srcInfo.UserDefined[ReservedMetadataPrefix+"compression"] = compressionAlgorithmV1
-		srcInfo.UserDefined[ReservedMetadataPrefix+"actual-size"] = strconv.FormatInt(actualSize, 10)
+		compressMetadata = make(map[string]string)
```
Comment (kannappanr) [inline cmd/object-handlers.go]: Since the number of keys is known, you can pre-allocate the memory.

`compressMetadata = make(map[string]string, 2)`

Comment (Praveenrajmani) [reply cmd/object-handlers.go]: done :+1:

**Review by harshavardhana — APPROVED** [review]

**Review by balamurugana — APPROVED** [review]

**Comment by nitisht** [comment]
blocking till mint-auto executes

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
Unblocked this PR can be merged @kannappanr

**Pull request merged**